logger/analytics/analyticslogger.go

```diff
@@ -127,6 +143,17 @@ func New(c Config) (*Logger, error) {
 		al.errLogger = logger.New(al.fhStream)
 	}
 
+	go func() {
+		for {
+			select {
+			case <-al.done:
+				return
+			case <-al.sendingTicker.C:
```

Comment:
Seems like this will force a flush every 10 minutes, even if there was a recent flush due to max size. Is that desirable? An alternative might be to use [time.Timer](https://golang.org/pkg/time/#Timer) and `timer.Reset()` if we flush via max-size, and when the timer goes off, create a new timer.

Reply:
The current solution *does* enforce that there will be a flush every 10 minutes or sooner. So in that sense, the code is fine. It just might not be exactly what a user would expect? Like if the logger typically fills up before 10 minutes, the current code will have some random small batches tossed in.

Reply:
Also from the "user perspective" I don't think it really matters. You're right that it would be submitting more frequently than they'd expect but from a "load on the server" perspective it should be relatively negligible.

And when it comes to reading from the stream, I'm pretty sure it doesn't matter how large the batches we write are since we're not compressing into aggregates or anything.

Reply:
Oops, just realized I didn't actually do this. I think for now I'll just punt on it in part due to laziness lol but in part due to it still feeling potentially tricky to get right (and not adding a ton of value).